modules/apps/foundation/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/categories/APIAuthenticationConfigurationCategory.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.configuration.admin.web.internal.categories;
+
+import com.liferay.configuration.admin.categories.ConfigurationCategory;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Jorge Ferrer
+ */
+@Component
+public class APIAuthenticationConfigurationCategory
+	implements ConfigurationCategory {
+
+	public static final String KEY = "api-authentication";
```

Comment:
Why do we make this field public and the other one private if we have getters for both fields?

Reply:
I did it so that developers could use APIAuthenticationConfigurationCategory.KEY in their Configuration interfaces in order to specify the desired category in the annotation. Even if there is a getter, that doesn't solve that need because it cannot be accessed from the annotation property.

The alternative is to just keep the string, like we have been doing up until now. Maybe it's not so bad.

This argument does not apply to _CATEGORY_SET_KEY since there is no need to reference that from outside.

Reply:
After thinking this through, I'm going to go ahead with your changes and just use the configuration key string in the Configuration interfaces annotations because otherwise I would need to create a dependency with configuration-admin-api, which I don't think is desirable.

Reply:
We can move the category keys to a `ConfigurationCategoryConstants` class in the api so devs can reference the category in their configuration interfaces. We've used the approach for the [form navigators](https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/servlet/taglib/ui/FormNavigatorConstants.java) and [screen navigation](https://github.com/liferay/liferay-portal/blob/master/modules/apps/foundation/users-admin/users-admin-api/src/main/java/com/liferay/users/admin/constants/UserFormConstants.java). What do you think?

Reply:
Yeah, I considered that as well. And actually got started by creating that class in configuration-admin-api.

However, that forced creating a dependency with the module from any module that has a Configuration interface. But that would mean that if configuration-admin-* is undeployed, all configurable modules would be undeployed too. And we definitely don't want that. Even without our own UI, the modules should work and can be configured through properties or even 3rd party tools like the Felix Web Console.

Makes sense?